ui/tests/e2e/fixtures/environmentsPage.js

```diff
@@ -0,0 +1,221 @@
+    await this.page.evaluate(() => {
+      const button = [...document.querySelectorAll('button')].find((b) =>
+        b.textContent.includes('Assigned Connections'),
+      );
+      if (button) {
+        button.setAttribute('data-testid', 'assigned-connections-button');
+      }
+    });
+
+    // Click the button using the newly added data-testid
+    await this.page.getByTestId('assigned-connections-button').click();
+
+    // Wait for and locate the dialog
+    const dialog = await this.page.locator('[aria-labelledby="alert-dialog-slide-title"]');
+    await expect(dialog).toBeVisible();
+    // Define the aria-labelledby values for checkboxes
+    const checkBoxAriaLabels = [
+      'transfer-list-item-kubernetes-mantra-9-label',
+      'transfer-list-item-meshery-themis-7-label',
+      'transfer-list-item-meshery-vader-9-label',
+      'transfer-list-item-meshery-exogorth-0-label',
+      'transfer-list-item-meshery-hydra-8-label',
+    ];
+
+    // Iterate over the aria-labels and check each checkbox
+    for (const label of checkBoxAriaLabels) {
+      const checkbox = dialog.locator(`input[aria-labelledby="${label}"]`);
+      await checkbox.waitFor({ state: 'visible' });
+      if (!(await checkbox.isChecked())) {
+        await checkbox.check();
+      }
+    }
+
+    await this.page.click('button[aria-label="move selected right"]');
+    await this.page.getByRole('button', { name: 'Save', exact: true }).isEnabled();
+    await this.page.getByRole('button', { name: 'Save', exact: true }).click();
+  }
+
+  async moveAssignedConnectionsToAvailable(environmentName) {
+    await expect(this.page).toHaveURL(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+    await this.page.waitForSelector(`text=${environmentName}`, { state: 'visible' });
+    await this.page.evaluate(() => {
+      const button = [...document.querySelectorAll('button')].find((b) =>
+        b.textContent.includes('Assigned Connections'),
+      );
+      if (button) {
+        button.setAttribute('data-testid', 'assigned-connections-button');
+      }
+    });
+
+    // Click the button using the newly added data-testid
+    await this.page.getByTestId('assigned-connections-button').click();
+
+    // Wait for and locate the dialog
+    const dialog = await this.page.locator('[aria-labelledby="alert-dialog-slide-title"]');
+    await expect(dialog).toBeVisible();
+
+    // Define the aria-labelledby values for checkboxes
+    const AssignedConnections = [
+      'transfer-list-item-kubernetes-mantra-9-label',
+      'transfer-list-item-meshery-themis-7-label',
+      'transfer-list-item-meshery-vader-9-label',
+      'transfer-list-item-meshery-exogorth-0-label',
+      'transfer-list-item-meshery-hydra-8-label',
+    ];
+
+    // Iterate over the aria-labels and check each checkbox in #rightList
+    for (const label of AssignedConnections) {
+      const checkbox = dialog.locator(`input[aria-labelledby="${label}"]`);
```

Comment:
Can we use a native playwright locator instead of .locator?

https://playwright.dev/docs/api/class-page#page-get-by-label
https://playwright.dev/docs/input#checkboxes-and-radio-buttons

Reply:
Yes @jerensl , we can definitely do that. The only catch is that we'll need to update the sistent repo since it's responsible for the modal. We'll need to merge the PR in that repo first, then follow up with this one.